api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -111,6 +111,10 @@ message OAuth2Config {
   // Forward the OAuth token as a Bearer to upstream web service.
   bool forward_bearer_token = 7;
 
+  // If set to true, preserve the existing authorization header when forward_bearer_token is false.
+  // Default value is false.
+  bool preserve_authorization_header = 16;
```

Comment:
Typically only one of `forward_bearer_token` and `preserve_authorization_header` could be set to true (?). I think it would be better to document this point.

Reply:
Agreed, and also think it might be worth mentioning the sanitization behavior here, something like `By default Envoy strips the authorization header before forwarding upstream.`